src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,279 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
-import styled from 'styled-components/native';
-import { TouchableOpacity, Text } from 'react-native';
+import styled, { ThemeContext } from 'styled-components/native';
+import { TouchableHighlight } from 'react-native';
 
-const StyledButton = styled(TouchableOpacity)(
-  (props) => `
-  background-color: ${props.theme.colors.shade[900]};
-  padding: 14px;
-  border-radius: 6px;
-  margin-top: 7px;
-`,
-);
+import Icon from '../Icon';
+import Space from '../Space';
+import Text from '../Text';
+import View from '../View';
+import Flex from '../Flex';
+import automation from '../../_helpers/automation-attributes';
+import { getColor, makePxValue, getVariantColorKeys } from '../../_helpers/theme';
+
+const styles = {
+  fontColor({ variant, variantColor, disabled }) {
```

Comment:
What if we just use `props` here instead of spreading?

Reply:
keeping it consistent as in other components.